src/main/java/org/opensearch/security/configuration/ConfigurationRepository.java

```diff
@@ -431,6 +431,32 @@ Future<Void> executeConfigurationInitialization(final SecurityMetadata securityM
         return CompletableFuture.completedFuture(null);
     }
 
+    public void createApiTokenIndex() {
+        try {
+            // wait for the cluster here until it will finish managed node election
+            while (clusterService.state().blocks().hasGlobalBlockWithStatus(RestStatus.SERVICE_UNAVAILABLE)) {
+                LOGGER.info("Wait for cluster to be available ...");
+                TimeUnit.SECONDS.sleep(1);
+            }
+
+            try {
+                final ThreadContext threadContext = threadPool.getThreadContext();
+                try (StoredContext ctx = threadContext.stashContext()) {
```

Comment:
Like the above, this is taken from the creation of `opendistro_security` index, which I assume has this to handle edge cases since the security index may not be initialized yet. Again not completely sure if its needed but I think it is fine?